components/amara/amara.app.mjs

```diff
@@ -0,0 +1,543 @@
+import { axios } from "@pipedream/platform";
+import constants from "./constants.mjs";
+
+export default {
+  type: "app",
+  app: "amara",
+  propDefinitions: {
+    team:	{
+      type: "string",
+      label: "Team",
+      description: "Team slug for the video or null to remove it from its team.",
+      optional: true,
+    },
+    videoUrl: {
+      type: "string",
+      label: "Video URL",
+      description: "The url for the video. Any url that Amara accepts will work here. You can send the URL for a file (e.g. `http:///www.example.com/my-video.ogv`), or a link to one of the accepted providers (YouTube, Vimeo).",
+    },
+    primaryAudioLanguageCode: {
+      type: "string",
+      label: "Primary audio language code",
+      description: "Language code for the main language spoken in the video.",
+      optional: true,
+      async options() {
+        const { languages } = await this.getLanguages();
+
+        return Object.keys(languages)
+          .map((key) => ({
+            label: languages[key],
+            value: key,
+          }));
+      },
+    },
+    project: {
+      type: "string",
+      label: "Project",
+      description: "Project slug for the video or null to put it in the default project",
+      optional: true,
+      async options({
+        prevContext, team,
+      }) {
+        const { url } = prevContext;
+        if (url === null || !team) {
+          return [];
+        }
+        const {
+          meta,
+          objects: projects,
+        } = await this.getTeamProjects({
+          team,
+        });
+        const options = projects.map((project) => ({
+          label: project.name,
+          value: project.slug,
+        }));
+        return {
+          options,
+          context: {
+            url: meta.next,
+          },
+        };
+      },
+    },
+    videoId: {
+      type: "string",
+      label: "Video ID",
+      description: "Video ID as the Amara video identifier",
+      async options({
+        prevContext, team,
+      }) {
+        const { url } = prevContext;
+
+        if (url === null) {
+          return [];
+        }
+
+        const {
+          meta,
+          objects: videos,
+        } = await this.listVideos({
+          url,
+          params: {
+            limit: 20,
+            team: team ?? null,
```

Comment:
Could we change this to `team: team || null` so we don't have to use [`utils.emptyStrToUndefined(team)`](https://github.com/PipedreamHQ/pipedream/blob/0ea1c26a13cd0183a9bfd8d731075de5d5293678/components/amara/actions/update-subtitle-language/update-subtitle-language.mjs#L23) in each action's `videoId` prop definition?